staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -147,3 +151,33 @@ func ComputeHistogramDelta(before, after model.Samples, label model.LabelName) {
 func makeKey(a, b model.LabelValue) string {
 	return string(a) + "___" + string(b)
 }
+
+// GetGaugeMetricValue extract metric value from GaugeMetric
+func GetGaugeMetricValue(t *testing.T, m metrics.GaugeMetric) float64 {
```

Comment:
+1 on @RainbowMango's comment. We should delete the `testing.T` parameter and return (float64, err) from the function instead. The caller can then use the error to `t.Errorf` if needed.